deploy/pg2sphinx_trigger.py

```diff
@@ -215,72 +235,99 @@ def pg_get_tables(sql_query,sql_db):
                         INSERT INTO indexes (
                             sphinx_index
                             , index_parent
-                            ,source
+                            , source
                             ) VALUES(?, ?, ?);""" ,(index.strip(), str(index_parent).strip(), source))
 
     # output
     sql = """
-        select
+        SELECT
             a.source as source
-            , coalesce(a.sql_db,b.sql_db) as database
+            , coalesce(a.sql_db, b.sql_db) as database
             , a.sql_query as sql
             , i.sphinx_index
             , i.index_parent
-            FROM 
-            indexes i left join indexes p on trim(i.index_parent)=trim(p.sphinx_index)
-            left join sources a on coalesce(i.source,p.source) = a.source
-            left join sources b on a.source_parent = b.source
+            FROM
+            indexes i LEFT JOIN indexes p on trim(i.index_parent) = trim(p.sphinx_index)
+            LEFT JOIN sources a on coalesce(i.source, p.source) = a.source
+            LEFT JOIN sources b on a.source_parent = b.source
     """
-    
-    resultat=[]
+
+    tables_target = []
+    db_path = options.database_filter.split('.')
+    is_table_target = bool(re.match('^[a-z]*_(dev|int|prod).[a-z]*.[a-z]*$', options.database_filter))
+    if is_table_target:
+        sql_target = 'SELECT * FROM %s.%s' % (db_path[1], db_path[2])
+        try:
+            tables_target = pg_get_tables(sql_target, db_path[0])
+        except Exception as e:
+            msg = 'Error while analysing table deps\n'
+            msg += 'SQL: %s\n%s' % (sql, e)
+            raise Exception(e)
+
+    results=[]
     for row in c.execute(sql):
         db = None
         indices = row['sphinx_index']
         indices_distributed = row['index_parent']
         # database filter
         # -d pattern
         if options.index_filter is None:
-            if (options.database_filter and options.database_filter.count('.')==0) or (options.database_filter == None):
+            if (options.database_filter and len(db_path) == 0) or (options.database_filter == None):
                 # db only filter can be applied to sphinx config, no need to query postgres db
                 if options.database_filter is None or options.database_filter == row['database']:
                     db = row['database']
                 # if db filter is more detailed, we have to analyze the sql queries with postgres ANALZYE VERBOSE
-            elif options.database_filter.split(".")[0] == row['database']:
-                table =  pg_get_tables(row['sql'], row['database'])
-                db = row['database'] if options.database_filter in table else None
+            elif db_path[0] == row['database']:
+                # In case the taget is table, only the SQL related to the target table for sphinx rotation indices
+                if is_table_target:
+                    if len([t for t in tables_target if t in row['sql']]) > 0:
```

Comment:
It s true, I would need to also add the table being deployed to the list created from explain analyze and create a set to remove duplicates.